external-crates/move/crates/move-compiler/tests/move_2024/parser/labeled_lambda_body_invalid.exp

```diff
@@ -1,3 +1,18 @@
+error[E04016]: too few arguments
+  ┌─ tests/move_2024/parser/labeled_lambda_body_invalid.move:7:9
+  │
+7 │         call!(|x| -> u64 'a: 0); // parsing error needs a block
+  │         ^^^^^^^^^^^^^^^^^^^^^^^
+  │         │    │
+  │         │    Found 0 argument(s) here
+  │         Invalid call of 'a::m::call'. The call expected 1 argument(s) but got 0
```

Comment:
Would it be worth it to build some machinery to avoid reporting these sorts of "double errors"? We already have an error, since we messed up parsing. Is it useful to report this second one, too?

If we decide not to, a few idea of how to avoid it:

(1) We keep track of bad-parse locations and, when reporting errors, compare the errors to see if it's the same problem (specifically look through the env diagnostics and see if we had a parsing error, and decide not to reporting a naming error at the same location). This could prove to be rather finnicky.

(2) We could mark the arguments as having errors (`vec[E::UnexpectedError]`) and check for that sort of thing when deciding to report errors later:
```
if args.length() != fn_args.length() && args.iter().any(|arg| arg.is_error) {
  .. report arity error
}
```

This would require hand-addressing these in a bunch of spots in the compiler.

(3) When reporting diagnostics, only report the first one we receive for each location. This would actually still produce this error, but not the one below it. That might be an improvement.

Alternatively, maybe we should just live with all these errors. We have, however, gotten some feedback that errors can be too noisy to find the root cause and I'm afraid this is going to make the problem worse.

This is possibly worth discussing as a group (cc @tnowacki )

Reply:
I don't think we even need to go that far, we just need to do a bit of error tuning. For this case, we could see if any of the call args contains an error and avoid reporting arity errors in that case.